Core/IntegrationTests/Api.cs

```diff
@@ -275,6 +274,59 @@ public async Task StreamGetBranches()
     Assert.That(res[0].name, Is.EqualTo("main"));
   }
 
+  [Test]
+  public async Task StreamGetBranches_Throws_WhenRequestingOverLimit()
+  {
+    Assert.ThrowsAsync<Exception>(
+      async () => await myClient.StreamGetBranches(streamId, ServerLimits.BRANCH_GET_LIMIT + 1).ConfigureAwait(false)
+    );
+    var res = await myClient.StreamGetBranches(streamId, ServerLimits.BRANCH_GET_LIMIT).ConfigureAwait(false);
+
+    Assert.That(res, Is.Not.Null);
+    Assert.That(res, Has.Count.EqualTo(ServerLimits.BRANCH_GET_LIMIT));
+  }
+
+  [Test]
+  public async Task StreamGetBranches_WithManyBranches()
+  {
+    var streamId = await myClient.StreamCreate(new StreamCreateInput { name = "Many branches stream" });
+
+    await CreateEmptyBranches(myClient, streamId, ServerLimits.BRANCH_GET_LIMIT);
+
+    var res = await myClient.StreamGetBranches(streamId, ServerLimits.BRANCH_GET_LIMIT);
+
+    Assert.That(res, Is.Not.Null);
+    Assert.That(res, Has.Count.EqualTo(ServerLimits.BRANCH_GET_LIMIT));
+  }
+
+  public async Task CreateEmptyBranches(
+    Client client,
+    string streamId,
+    int branchCount,
+    string branchPrefix = "Test branch"
+  )
+  {
+    // now let's send HTTP requests to each of these URLs in parallel
+    var options = new ParallelOptions { MaxDegreeOfParallelism = 20 };
+
+    // now let's send HTTP requests to each of these URLs in parallel
+    await Parallel.ForEachAsync(
+      Enumerable.Range(0, branchCount),
+      options,
+      async (i, cancellationToken) =>
+      {
+        await client.BranchCreate(
```

Comment:
Are the branches literally created on the server or are they created as a mock of some sort. I don't think we want to create 500 fresh branches each time do we?

Reply:
If this DOES create branches we should tidy them up, maybe us a particular naming convention also, so another test could check that we don't have a bazillon branches and we fail the test, forcing us to do clean up or something, just thinking out loud

Reply:
Yes it does create 500 branches, but the server we're using is a brand new one spun up for every CI run. So cleanup happens once the job is over. No deployed servers are used for integration testing.

That said, I do agree with the sentiment, but our current way of doing these tests doesn't allow for much flexibility.

Reply:
No that's perfect